[Docs] Core Cog Guide (#4597)
Cog-Creators/Red-DiscordBot: docs/cog_guides/core.rst

.. _core:

====
Core
====

This is the cog guide for the core cog. You will
find detailed docs about usage and commands.

``[p]`` is considered as your prefix.

.. note:: To use this cog, load it by typing this::

        [p]load core

> 

Reviewer: Core does not need to be loaded, this part should be omitted from this guide

**Description**

Toggle the user's embed setting for DMs.

If enabled is None, the setting will be unset and
the global default will be used instead.

If set, this is used instead of the global default
to determine whether or not to use embeds. This is
used for all commands executed in a DM with the bot.

Examples:
    - ``[p]embedset user False`` - Disables embeds in your DMs.
    - ``[p]embedset user`` - Resets value to use global default.

**Arguments:**

- ``[enabled]`` Whether to use embeds in your DMs. Leave blank to reset to default.

.. _core-command-helpset:

^^^^^^^
helpset
^^^^^^^

.. note:: |owner-lock|

**Syntax**

.. code-block:: none

    [p]helpset 

**Description**

Commands to manage settings for the help command.

All help settings are applied globally.

.. _core-command-helpset-deletedelay:

"""""""""""""""""""
helpset deletedelay
"""""""""""""""""""

**Syntax**

.. code-block:: none

    [p]helpset deletedelay <seconds>

**Description**

Set the delay after which help pages will be deleted.

The setting is disabled by default, and only applies to non-menu help,
sent in server text channels.
Setting the delay to 0 disables this feature.

> The bot has to have MANAGE_MESSAGES permission for this to work.

Reviewer: I don't believe this is true, if the bot is just deleting its own messages

Author: This is based on the decorator `@commands.bot_has_permissions(manage_messages=True)` for that command
I definitely see why it shouldn't be necessary to delete your own messages though. If the decorator could be removed, then it'd make sense to remove it from the docs.

Reviewer: It appears to be there to allow the bot to use the mass delete endpoint. So the permission is required.

""""""""""""""""""""""
helpset resetformatter
""""""""""""""""""""""

**Syntax**

.. code-block:: none

    [p]helpset resetformatter 

**Description**

This resets Red's help formatter to the default formatter.

.. _core-command-helpset-resetsettings:

"""""""""""""""""""""
helpset resetsettings
"""""""""""""""""""""

**Syntax**

.. code-block:: none

    [p]helpset resetsettings 

**Description**

This resets Red's help settings to their defaults.

This may not have an impact when using custom formatters from 3rd party cogs

.. _core-command-helpset-showaliases:

"""""""""""""""""""
helpset showaliases
"""""""""""""""""""

**Syntax**

.. code-block:: none

    [p]helpset showaliases [show_aliases]

**Description**

This allows the help command to show existing commands aliases if there is any.

This defaults to True.
Using this without a setting will toggle.

.. _core-command-helpset-showhidden:

""""""""""""""""""
helpset showhidden
""""""""""""""""""

**Syntax**

.. code-block:: none

    [p]helpset showhidden [show_hidden]

**Description**

This allows the help command to show hidden commands.

This defaults to False.
Using this without a setting will toggle.

Examples:

>     - ``[p]helpset showhidden False`` - Disables embeds on this server.

Reviewer: I think you forgot to edit this line when you copy pasted :P

Author: Looks like I missed a lot of lines in this section actually. Ended up making more changes than you asked for in this section, including adding docs for the new `showaliases` command.

    - ``[p]helpset showhidden`` - Toggles the value.

**Arguments:**

- ``[show_hidden]`` Whether to use embeds on this server. Leave blank to toggle.

.. _core-command-helpset-showsettings:

""""""""""""""""""""
helpset showsettings
""""""""""""""""""""

**Syntax**

.. code-block:: none

    [p]helpset showsettings 

**Description**

Show the current help settings.

> .. Warning:: These setting may not be accurate if the default formatter is not in use.

Reviewer: ```suggestion
.. Warning:: These settings may not be accurate if the default formatter is not in use.
```

.. _core-command-helpset-tagline:

"""""""""""""""
helpset tagline
"""""""""""""""

**Syntax**

.. code-block:: none

    [p]helpset tagline [tagline]

**Description**

Set the tagline to be used.

The maximum tagline length is 2048 characters.
This setting only applies to embedded help. If no tagline is specified, the default will be used instead.

Examples:
    - ``[p]helpset tagline Thanks for using the bot!``
    - ``[p]helpset tagline`` - Resets the tagline to the default.

**Arguments:**

- ``[tagline]`` The tagline to appear at the bottom of help embeds. Leave blank to reset.

.. _core-command-helpset-usemenus:

""""""""""""""""
helpset usemenus
""""""""""""""""

**Syntax**

.. code-block:: none

    [p]helpset usemenus [use_menus]

**Description**

Allows the help command to be sent as a paginated menu instead of separate
messages.

> When enabled, ``[p]help`` will only show one page at a time and use reactions to navigate to other pages.

Reviewer: ```suggestion
When enabled, ``[p]help`` will only show one page at a time and will use reactions to navigate between pages.
```

This defaults to False.
Using this without a setting will toggle.

 Examples:
    - ``[p]helpset usemenues False`` - Disables using menus on this server.
    - ``[p]helpset usemenues`` - Toggles the value.

**Arguments:**

- ``[use_menus]`` Whether to use menus on this server. Leave blank to toggle.

.. _core-command-helpset-usetick:

"""""""""""""""
helpset usetick
"""""""""""""""

**Syntax**

.. code-block:: none

    [p]helpset usetick [use_tick]

**Description**

This allows the help command message to be ticked if help is sent in a DM.

Defaults to False.
Using this without a setting will toggle.

.. Note:: This is only used when the bot is not using menus.

Examples:

>     - ``[p]helpset usetick False`` - Disables ticking in DMs.

Reviewer: ```suggestion
    - ``[p]helpset usetick False`` - Disables ticking when help is sent in DMs.
```

Author: Rewording this to `Disables ticking when help is sent to DMs.` to clarify it's talking about where the response from the bot ends up. (i.e. it ticks when I use `[p]help` in a server but the results are sent to my DMs)

    - ``[p]helpset usetick`` - Toggles the value.

**Arguments:**

> - ``[use_tick]`` Whether to tick the help command in DMs. Leave blank to toggle.

Reviewer: ```suggestion
- ``[use_tick]`` Whether to tick the help command when help is sent in DMs. Leave blank to toggle.
```

Author: Same as the above, rewording to `Whether to tick the help command when help is sent to DMs`

.. _core-command-helpset-verifychecks:

""""""""""""""""""""
helpset verifychecks
""""""""""""""""""""

**Syntax**

.. code-block:: none

    [p]helpset verifychecks [verify]

**Description**

Sets if commands which can't be run in the current context should be filtered from help.

Defaults to True.
Using this without a setting will toggle.

Examples:
    - ``[p]helpset verifychecks False`` - Enables showing unusable commands.
    - ``[p]helpset verifychecks`` - Toggles the value.

**Arguments:**

- ``[verify]`` Whether to hide unusable commands in help. Leave blank to toggle.

.. _core-command-helpset-verifyexists:

""""""""""""""""""""
helpset verifyexists
""""""""""""""""""""

**Syntax**

.. code-block:: none

    [p]helpset verifyexists [verify]

**Description**

> This allows the bot to respond indicating the existence of a specific help topic even if the user can't use it.

Reviewer: I believe the point of this help setting is to let the user know when a command does *not* exist, ie `[p]help pinh` will respond "Command `pinh` does not exist."

Author: This was the existing description for the command. I will look into it further and get back to this one.

Author: I believe it is accurate as written. 

As an example:
With `verifyexists` = False, my bot will not respond to `[p]help cleanup duplicates` on the Red support server since it does not have manage messages permissions.
With `verifyexists` = True, my bot will respond to `[p]help cleanup duplicates` despite not being able to use it.

Author: This is now being handled in #4892

.. Note:: This setting on it's own does not fully prevent command enumeration.

Defaults to False.
Using this without a setting will toggle.

Examples:

>     - ``[p]helpset verifyexists False`` - Enables showing unusable commands.

Reviewer: This description is not accurate

Reviewer: This has still not been fixed. This is unrelated to the other issue, as that text is describing a different behavior which IS accurate (although potentially not intended).

Author: This is now resolved, along with a better description of the command.

Examples:
    - ``[p]ignore channel #general`` - Ignores commands in the #general channel.
    - ``[p]ignore channel`` - Ignores commands in the current channel.
    - ``[p]ignore channel "General Channels"`` - Use quotes for categories with spaces.
    - ``[p]ignore channel 356236713347252226`` - Also accepts IDs.

**Arguments:**

- ``<channel>`` The channel to ignore. Can be a category channel.

.. _core-command-ignore-list:

"""""""""""
ignore list
"""""""""""

**Syntax**

.. code-block:: none

    [p]ignore list 

**Description**

List the currently ignored servers and channels.

Example:
    - ``[p]ignore list``

.. _core-command-ignore-server:

"""""""""""""
ignore server
"""""""""""""

.. note:: |admin-lock|

**Syntax**

.. code-block:: none

    [p]ignore server 

.. tip:: Alias: ``ignore guild``

**Description**

Ignore commands in this server.

.. Note:: Owners, Admins, and those with Manage Server permissions override ignored servers.

Example:
    - ``[p]ignore server`` - Ignores the current server

.. _core-command-info:

^^^^
info
^^^^

**Syntax**

.. code-block:: none

    [p]info 

**Description**

Shows info about Red.

> See ``[p]custominfo`` to customize.

Reviewer: ```suggestion
See ``[p]set custominfo`` to customize.
```

.. _core-command-invite:

^^^^^^
invite
^^^^^^

**Syntax**

.. code-block:: none

    [p]invite 

**Description**

Shows Red's invite url.

This will always send the invite to DMs to keep it private.

This command is locked to the owner unless ``[p]inviteset public`` is set to True.

.. _core-command-inviteset:

^^^^^^^^^
inviteset
^^^^^^^^^

.. note:: |owner-lock|

**Syntax**

.. code-block:: none

    [p]inviteset 

**Description**

Commands to setup Red's invite settings.

.. _core-command-inviteset-perms:

"""""""""""""""
inviteset perms
"""""""""""""""

**Syntax**

.. code-block:: none

    [p]inviteset perms <level>

**Description**

Make the bot create its own role with permissions on join.

The bot will create its own role with the desired permissions        when it joins a new server. This is a special role that can't be        deleted or removed from the bot.

For that, you need to provide a valid permissions level.
You can generate one here: https://discordapi.com/permissions.html

> Please note that you might need two factor authentication for        some permissions.

Reviewer: This section was probably formatted strangely in the docstring, as it is formatted wrong here

Example:
    - ``[p]inviteset perms 134217728`` - Adds a "Manage Nicknames" permission requirement to the invite.

**Arguments:**

- ``<level>`` The permission level to require for the bot in the generated invite.

.. _core-command-inviteset-public:

""""""""""""""""
inviteset public
""""""""""""""""

**Syntax**

.. code-block:: none

    [p]inviteset public [confirm=False]

**Description**

Toggles if ``[p]invite`` should be accessible for the average user.

> Bot must me made into a ``Public bot`` in the developer dashboard for public invites to work.

Reviewer: ```suggestion
The bot must be made into a ``Public bot`` in the developer dashboard for public invites to work.
```

    [p]licenseinfo 

.. tip:: Alias: ``licenceinfo``

**Description**

Get info about Red's licenses.

.. _core-command-load:

^^^^
load
^^^^

.. note:: |owner-lock|

**Syntax**

.. code-block:: none

    [p]load <cogs...>

**Description**

Loads cog packages from the local paths and installed cogs.

See packages available to load with ``[p]cogs``.

Additional cogs can be added using Downloader, or from local paths using ``[p]addpath``.

Examples:
    - ``[p]load general`` - Loads the ``general`` cog.
    - ``[p]load admin mod mutes`` - Loads multiple cogs.

**Arguments:**

- ``<cogs...>`` The cog packages to load.

.. _core-command-localallowlist:

^^^^^^^^^^^^^^
localallowlist
^^^^^^^^^^^^^^

.. note:: |admin-lock|

**Syntax**

.. code-block:: none

    [p]localallowlist 

.. tip:: Alias: ``localwhitelist``

**Description**

Commands to manage the server specific allowlist.

.. Warning:: When the allowlist is in use, the bot will ignore commands from everyone not on the list in the server.

Use ``[p]localallowlist clear`` to disable the allowlist

.. _core-command-localallowlist-add:

""""""""""""""""""
localallowlist add
""""""""""""""""""

**Syntax**

.. code-block:: none

>     [p]localallowlist add <users_or_roles...>

Reviewer: Here you use `<users_or_roles...>`, but below has only `<users...>`

Author: Adjusted all to be `<users_or_roles...>`

**Description**

Adds a user or role to the server allowlist.

Examples:
    - ``[p]localallowlist add @26 @Will`` - Adds two users to the local allowlist
    - ``[p]localallowlist add 262626262626262626`` - Adds a user by ID

**Arguments:**

- ``<users...>`` The user or users to remove from the local allowlist.

.. _core-command-localallowlist-clear:

""""""""""""""""""""
localallowlist clear
""""""""""""""""""""

**Syntax**

.. code-block:: none

    [p]localallowlist clear 

**Description**

Clears the allowlist.

This disables the local allowlist.

Example:
    - ``[p]localallowlist clear``

.. _core-command-localallowlist-list:

"""""""""""""""""""
localallowlist list
"""""""""""""""""""

**Syntax**

.. code-block:: none

    [p]localallowlist list 

**Description**

Lists users and roles on the server allowlist.

Example:
    - ``[p]localallowlist list``

.. _core-command-localallowlist-remove:

"""""""""""""""""""""
localallowlist remove
"""""""""""""""""""""

**Syntax**

.. code-block:: none

>     [p]localallowlist remove <users_or_roles...>

Reviewer: Same issue as above

**Description**

Removes user or role from the allowlist.

The local allowlist will be disabled if all users are removed.

Examples:
    - ``[p]localallowlist remove @26 @Will`` - Removes two users from the local allowlist
    - ``[p]localallowlist remove 262626262626262626`` - Removes a user by ID

**Arguments:**

- ``<users...>`` The user or users to remove from the local allowlist.

.. _core-command-localblocklist:

^^^^^^^^^^^^^^
localblocklist
^^^^^^^^^^^^^^

.. note:: |admin-lock|

**Syntax**

.. code-block:: none

    [p]localblocklist 

.. tip:: Alias: ``localblacklist``

**Description**

Commands to manage the server specific blocklist.

Use ``[p]localblocklist clear`` to disable the blocklist

.. _core-command-localblocklist-add:

""""""""""""""""""
localblocklist add
""""""""""""""""""

**Syntax**

.. code-block:: none

>     [p]localblocklist add <users_or_roles...>

Reviewer: Same issue as above

**Description**

Adds a user or role to the local blocklist.

Examples:
    - ``[p]blocklist add @26 @Will`` - Adds two users to the local blocklist
    - ``[p]blocklist add 262626262626262626`` - Adds a user by ID

**Arguments:**

- ``<users...>`` The user or users to add to the local blocklist.

.. _core-command-localblocklist-clear:

""""""""""""""""""""
localblocklist clear
""""""""""""""""""""

**Syntax**

.. code-block:: none

    [p]localblocklist clear 

**Description**

Clears the server blocklist.

Example:
    - ``[p]blocklist clear``

.. _core-command-localblocklist-list:

"""""""""""""""""""
localblocklist list
"""""""""""""""""""

**Syntax**

.. code-block:: none

    [p]localblocklist list 

**Description**

> Lists users and roles on the blocklist.

Reviewer: ```suggestion
Lists users and roles on the server blocklist.
```

Example:
    - ``[p]localblocklist list``

.. _core-command-localblocklist-remove:

"""""""""""""""""""""
localblocklist remove
"""""""""""""""""""""

**Syntax**

.. code-block:: none

>     [p]localblocklist remove <users_or_roles...>

Reviewer: Same issue as above

**Description**

Removes user or role from blocklist.

Examples:
    - ``[p]localblocklist remove @26 @Will`` - Removes two users from the local blocklist
    - ``[p]localblocklist remove 262626262626262626`` - Removes a user by ID

**Arguments:**

- ``<users...>`` The user or users to remove from the local blocklist.

.. _core-command-mydata:

^^^^^^
mydata
^^^^^^

**Syntax**

.. code-block:: none

    [p]mydata 

**Description**

Commands which interact with the data Red has about you.

More information can be found in the :doc:`End User Data Documentation.<../red_core_data_statement>`

.. _core-command-mydata-3rdparty:

"""""""""""""""
mydata 3rdparty
"""""""""""""""

**Syntax**

.. code-block:: none

    [p]mydata 3rdparty 

**Description**

View the End User Data statements of each 3rd-party module.

> This will send an attachment with the End User Data statements of all loaded 3rd party cog.

Reviewer: ```suggestion
This will send an attachment with the End User Data statements of all loaded 3rd party cogs.
```

**Description**

[Coming Soon] Get what data Red has about you.

.. _core-command-mydata-ownermanagement:

""""""""""""""""""""""
mydata ownermanagement
""""""""""""""""""""""

.. note:: |owner-lock|

**Syntax**

.. code-block:: none

    [p]mydata ownermanagement 

**Description**

Commands for more complete data handling.

.. _core-command-mydata-ownermanagement-allowuserdeletions:

"""""""""""""""""""""""""""""""""""""""""
mydata ownermanagement allowuserdeletions
"""""""""""""""""""""""""""""""""""""""""

**Syntax**

.. code-block:: none

    [p]mydata ownermanagement allowuserdeletions 

**Description**

Set the bot to allow users to request a data deletion.

This is on by default.
Opposite of ``[p]mydata ownermanagement disallowuserdeletions``

.. _core-command-mydata-ownermanagement-deleteforuser:

""""""""""""""""""""""""""""""""""""
mydata ownermanagement deleteforuser
""""""""""""""""""""""""""""""""""""

**Syntax**

.. code-block:: none

    [p]mydata ownermanagement deleteforuser <user_id>

**Description**

Delete data Red has about a user for a user.

This will cause the bot to get rid of or disassociate a lot of non-operational data from the specified user.

> Users have access to different command for this unless they can't interact with the bot at all.

Reviewer: ```suggestion
Users have access to a different command for this unless they can't interact with the bot at all.
```

mydata ownermanagement setuserdeletionlevel
"""""""""""""""""""""""""""""""""""""""""""

**Syntax**

.. code-block:: none

    [p]mydata ownermanagement setuserdeletionlevel <level>

**Description**

Sets how user deletions are treated.

Example:
    - ``[p]mydata ownermanagement setuserdeletionlevel 1``

**Arguments:**

- ``<level>`` The strictness level for user deletion. See Level guide below.

Level:
    - ``0``: What users can delete is left entirely up to each cog.
    - ``1``: Cogs should delete anything the cog doesn't need about the user.

.. _core-command-mydata-whatdata:

"""""""""""""""
mydata whatdata
"""""""""""""""

**Syntax**

.. code-block:: none

    [p]mydata whatdata 

**Description**

Find out what type of data Red stores and why.

.. _core-command-reload:

^^^^^^
reload
^^^^^^

.. note:: |owner-lock|

**Syntax**

.. code-block:: none

    [p]reload <cogs...>

**Description**

Reloads cog packages.

This will unload and then load the specified cogs.

Cogs that were not loaded will be loaded.

Examples:
    - ``[p]reload general`` - Unloads then loads the ``general`` cog.
    - ``[p]reload admin mod mutes`` - Unloads then loads multiple cogs.

**Arguments:**

> - ``<cogs...>`` The cog packages to unload.

Reviewer: ```suggestion
- ``<cogs...>`` The cog packages to reload.
```

 - ``[p]addrole``
 - ``[p]ban``
 - ``[p]ignore guild``

 And more.

 Examples:
    - ``[p]set addadminrole @Admins``
    - ``[p]set addadminrole Super Admins``

**Arguments:**

- ``<role>`` The role to add as an admin.

.. _core-command-set-addmodrole:

""""""""""""""
set addmodrole
""""""""""""""

.. note:: |guildowner-lock|

**Syntax**

.. code-block:: none

    [p]set addmodrole <role>

**Description**

Adds a moderator role for this guild.

This grants access to moderator level commands like:
 - ``[p]mute``
 - ``[p]cleanup``
 - ``[p]customcommand create``

 And more.

 Examples:
    - ``[p]set addmodrole @Mods``
    - ``[p]set addmodrole Loyal Helpers``

**Arguments:**

- ``<role>`` The role to add as a moderator.

.. _core-command-set-api:

"""""""
set api
"""""""

.. note:: |owner-lock|

**Syntax**

.. code-block:: none

    [p]set api <service> <tokens>

**Description**

Commands to set, list or remove various external API tokens.

This setting will be asked for by some 3rd party cogs and some core cogs.

To add the keys provide the service name and the tokens as a comma separated
list of key,values as described by the cog requesting this command.

> .. Note:: API tokens are sensitive and should only be used in a private channel or in DM with the bot.

Reviewer: ```suggestion
.. Note:: API tokens are sensitive, so this command should only be used in a private channel or in DM with the bot.
```

    - ``[p]set competing London 2012 Olympic Games``

**Arguments:**

- ``[competing]`` The text to follow ``Competing in``. Leave blank to clear the current activity status.

.. _core-command-set-custominfo:

""""""""""""""
set custominfo
""""""""""""""

.. note:: |owner-lock|

**Syntax**

.. code-block:: none

    [p]set custominfo [text]

**Description**

Customizes a section of ``[p]info``.

The maximum amount of allowed characters is 1024.
Supports markdown, links and "mentions".

Link example: ``[My link](https://example.com)``

Examples:
    - ``[p]set custominfo >>> I can use **markdown** such as quotes, ||spoilers|| and multiple lines``
    - ``[p]set custominfo Join my [support server](discord.gg/discord)``
    - ``[p]set custominfo`` - Removes custom info text.

**Arguments:**

- ``[text]`` The custom info text.

.. _core-command-set-deletedelay:

"""""""""""""""
set deletedelay
"""""""""""""""

.. note:: |guildowner-lock|

**Syntax**

.. code-block:: none

    [p]set deletedelay [time]

**Description**

Set the delay until the bot removes the command message.

Must be between -1 and 60.

Set to -1 to disable this feature.

> This is applied only the current server and not globally.

Reviewer: ```suggestion
This is only applied to the current server and not globally.
```

Examples:
    - ``[p]set deletedelay`` - Shows the current delete delay setting.
    - ``[p]set deletedelay 60`` - Sets the delete delay to the max of 60 seconds.
    - ``[p]set deletedelay -1`` - Disables deleting command messages.

**Arguments:**

- ``[time]`` The seconds to wait before deleting the command message. Use -1 to disable.

.. _core-command-set-description:

"""""""""""""""
set description
"""""""""""""""

.. note:: |owner-lock|

**Syntax**

.. code-block:: none

    [p]set description [description]

**Description**

Sets the bot's description.
Use without a description to reset.
This is shown in a few locations, including the help menu.

The maximum description length is 250 characters to ensure it displays properly.

The default is "Red V3".

Examples:
    - ``[p]set description`` - Resets the description to the default setting.
    - ``[p]set description MyBot: A Red V3 Bot``

**Arguments:**

> - ``[time]`` The seconds to wait before deleting the command message. Use -1 to disable.

Reviewer: This shouldn't be here

Author: Added the proper argument name and description.

.. note:: |admin-lock|

**Syntax**

.. code-block:: none

    [p]set serverprefix [prefixes...]

.. tip:: Alias: ``set serverprefixes``

**Description**

Sets Red's server prefix(es).

.. Warning:: This will override global prefixes, the bot will not respond to any global prefixes in this server.

    This is not additive. It will replace all current server prefixes.

Examples:
    - ``[p]set serverprefix !``
    - ``[p]set serverprefix "@Red "`` - Quotes are needed to use spaces. This uses a mention as the prefix.
    - ``[p]set serverprefix ! ? .`` - Sets multiple prefixes.

**Arguments:**

- ``[prefixes...]`` The prefixes the bot will respond to on this server. Leave blank to clear server prefixes

.. _core-command-set-showsettings:

""""""""""""""""
set showsettings
""""""""""""""""

**Syntax**

.. code-block:: none

    [p]set showsettings 

**Description**

Show the current settings for Red.

.. _core-command-set-status:

""""""""""
set status
""""""""""

.. note:: |owner-lock|

**Syntax**

.. code-block:: none

    [p]set status <status>

**Description**

Sets Red's status.

Available statuses:
    - ``online``
    - ``idle``
    - ``dnd``
    - ``invisible``

Examples:

>     - ``[p]set status online`` - Clears the activity status.

Reviewer: ```suggestion
    - ``[p]set status online`` - Clears the status.
```

    - ``[p]set status invisible``

**Arguments:**

- ``<status>`` One of the available statuses.

.. _core-command-set-streaming:

"""""""""""""
set streaming
"""""""""""""

.. note:: |owner-lock|

**Syntax**

.. code-block:: none

    [p]set streaming [(<streamer> <stream_title>)]

.. tip:: Aliases: ``set stream``, ``set twitch``

**Description**

Sets Red's streaming status to a twitch stream.

This will appear as ``Streaming <stream_title>`` or ``LIVE ON TWITCH`` depending on the context.
It will also include a ``Watch`` button with a twitch.tv url for the provided streamer.

Maximum length for a stream title is 128 characters.

Leaving both streamer and stream_title empty will clear it.

Examples:
    - ``[p]set competing`` - Clears the activity status.

>     - ``[p]set competing London 2012 Olympic Games``

Reviewer: Incorrect

Author: Added correct examples here.
Also adjusted the arguments to be in `<>` to indicate required, since that's how each individual argument is presented in the usage.

This will appear as ``Watching <watching>``.

Maximum length for a watching status is 128 characters.

Examples:
    - ``[p]set watching`` - Clears the activity status.
    - ``[p]set watching [p]help``

**Arguments:**

- ``[watching]`` The text to follow ``Watching``. Leave blank to clear the current activity status.

.. _core-command-shutdown:

^^^^^^^^
shutdown
^^^^^^^^

.. note:: |owner-lock|

**Syntax**

.. code-block:: none

    [p]shutdown [silently=False]

**Description**

Shuts down the bot.

Allows Red to shut down gracefully.

This is the recommended method for shutting down the bot.

.. _core-command-traceback:

^^^^^^^^^
traceback
^^^^^^^^^

.. note:: |owner-lock|

**Syntax**

.. code-block:: none

    [p]traceback [public=False]

**Description**

Sends to the owner the last command exception that has occurred.

If public (yes is specified), it will be sent to the chat instead.

.. Warning:: Sending the traceback publicly can accidentally reveal sensitive information about your computer or configuration.

Examples:
    - ``[p]traceback`` - Sends the traceback to your DMs.
    - ``[p]traceback True`` - Sends the last traceback in the current context.

**Arguments:**

> - ``[public]`` Whether to send traceback to the current context. Leave blank to send to DMs.

Reviewer: ```suggestion
- ``[public]`` Whether to send the traceback to the current context. Leave blank to send to your DMs.
```

.. _core-command-unignore:

^^^^^^^^
unignore
^^^^^^^^

.. note:: |admin-lock|

**Syntax**

.. code-block:: none

    [p]unignore 

**Description**

Commands to remove servers or channels from the ignore list.

.. _core-command-unignore-channel:

""""""""""""""""
unignore channel
""""""""""""""""

**Syntax**

.. code-block:: none

    [p]unignore channel [channel]

**Description**

Remove a channel or category from the ignore list.

Defaults to the current channel.

Examples:
    - ``[p]unignore channel #general`` - Unignores commands in the #general channel.
    - ``[p]unignore channel`` - Unignores commands in the current channel.
    - ``[p]ignore channel "General Channels"`` - Use quotes for categories with spaces.

>     - ``[p]unignore channel 356236713347252226`` - Also accepts IDs. Use this method to ignore categories.

Reviewer: ```suggestion
    - ``[p]unignore channel 356236713347252226`` - Also accepts IDs. Use this method to unignore categories.
```

Author: Accepted and an additional ignore needed to be unignore on 3321

**Arguments:**

> - ``<channel>`` The channel to ignore. Can be a category channel.

Reviewer: ```suggestion
- ``<channel>`` The channel to unignore. Can be a category channel.
```

.. _core-command-unignore-server:

"""""""""""""""
unignore server
"""""""""""""""

.. note:: |admin-lock|

**Syntax**

.. code-block:: none

    [p]unignore server 

.. tip:: Alias: ``unignore guild``

**Description**

Remove this server from the ignore list.

Example:

>     - ``[p]unignore server`` - Ignores the current server

Reviewer: ```suggestion
    - ``[p]unignore server`` - Stops ignoring the current server
```
